src/puffin/src/puffin_manager/stager/bounded_stager.rs

```diff
@@ -284,6 +294,14 @@ impl Stager for BoundedStager {
 
         Ok(())
     }
+
+    async fn purge(&self, puffin_file_name: &str) -> Result<()> {
+        let file_name = puffin_file_name.to_string();
+        self.cache
+            .invalidate_entries_if(move |_k, v| v.puffin_file_name() == file_name)
+            .unwrap(); // SAFETY: `support_invalidation_closures` is enabled
```

Comment:
Can we call `self.cache.run_pending_tasks()` here? `invalidate_entries_if()` only sets the callback. We can execute the maintenance task here.